### ISSUE TITLE
Software design document

### ISSUE THREAD

**Issue by jan-gogo** [Flank/flank#1536]
# Description

This is an SDD template proposal.  

## References

* https://en.wikipedia.org/wiki/Software_design_description
* https://blog.tara.ai/software-design-documents/
* https://github.com/platform-platform/monorepo/blob/master/docs/00_template.md

## Motivation

Is important to have clear communication between product owner and developer. 
The SDD can be a bridge between business vision and low-level implementation.
The lack of SDD can lead to misunderstanding and waste.

## Proposed solution

Prepare a template that will help developers preparing consistent SDD, without a doubt.

The metrics project already has well defined [SDD template](https://github.com/platform-platform/monorepo/blob/master/docs/00_template.md) that meets our requirements, so it will be convenient to reuse it also for flank possibly after some adjustments.

**Comment by jan-gogo** [comment]
# Project name

> Summary of the proposed change

# References

> Link to supporting documentation, GitHub tickets, etc.

# Motivation

> What problem is this project solving?

# Goals

> Identify success metrics and measurable goals.

# Non-Goals

> Identify what's not in scope.

# Design

> Explain and diagram the technical design
>
> Identify risks and edge cases

# API

> What will the proposed API look like?

# Dependencies

> What is the project blocked on?

> What will be impacted by the project?

# Testing

> How will the project be tested?

# Alternatives Considered

> Summarize alternative designs (pros & cons)

# Timeline

> Document milestones and deadlines.

DONE:

  -

NEXT:

  -
  
# Results

> What was the outcome of the project?
